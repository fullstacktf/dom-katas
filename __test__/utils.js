import fs from "fs";

export function readRelativeIndex(dirname) {
  const html = fs.readFileSync(dirname + "/../index.html", "utf8");

  const body = html.split(/<body .*>/)[1];

  const bodyContent = body.split("</body>")[0];

  const data = bodyContent.replace(/<script .*>.*<\/script>/, "");

  return data;
}
