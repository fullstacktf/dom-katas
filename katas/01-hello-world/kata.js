export function kata() {
  const div = document.createElement("div");
  div.id = "square";
  const p = document.createElement("p");
  p.id = "text";
  p.textContent = "Hello World";

  div.appendChild(p);

  document.body.appendChild(div);
}
