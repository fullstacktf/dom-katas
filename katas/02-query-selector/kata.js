export function kata() {
  const starWars = document.querySelector("main > section:nth-child(2) > p");
  starWars.classList.add("star-wars");
}
