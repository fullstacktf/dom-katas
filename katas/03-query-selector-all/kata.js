export function kata() {
  const titles = document.querySelectorAll("section > h2");

  for (const title of titles) {
    title.classList.add("title");
  }
}
