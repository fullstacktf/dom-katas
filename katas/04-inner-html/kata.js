const page = `<main>
<article>
    <img src="https://picsum.photos/id/28/800/300">
    <div class="content">
        <p class="category">travel</p>
        <h2>Trip Planning for Total Beginners</h2>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.</p>
    </div>
</article>
<article>
    <img src="https://picsum.photos/id/35/800/300">
    <div class="content">
        <p class="category">travel</p>
        <h2>7 ways to survive in the desert</h2>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.</p>
    </div>
</article>
<article>
    <img src="https://picsum.photos/id/37/800/300">
    <div class="content">
        <p class="category">travel</p>
        <h2>5 amazing travel destinations</h2>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.</p>
    </div>
</article>
</main>`;

export function kata() {
  document.body.innerHTML = page;
}
