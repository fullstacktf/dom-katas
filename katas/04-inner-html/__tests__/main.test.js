import { getByText, prettyDOM } from "@testing-library/dom";
import { readRelativeIndex } from "../../../__test__/utils.js";

import { kata } from "../kata.js";

const html = readRelativeIndex(__dirname);

test("structure is ok", async () => {
  document.body.innerHTML = html;
  kata();

  expect(document.body).toMatchSnapshot();
});
