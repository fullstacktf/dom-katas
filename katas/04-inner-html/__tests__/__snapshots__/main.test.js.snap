// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`structure is ok 1`] = `
<body>
  <main>
    

    <article>
      
    
      <img
        src="https://picsum.photos/id/28/800/300"
      />
      
    
      <div
        class="content"
      >
        
        
        <p
          class="category"
        >
          travel
        </p>
        
        
        <h2>
          Trip Planning for Total Beginners
        </h2>
        
        
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.
        </p>
        
    
      </div>
      

    </article>
    

    <article>
      
    
      <img
        src="https://picsum.photos/id/35/800/300"
      />
      
    
      <div
        class="content"
      >
        
        
        <p
          class="category"
        >
          travel
        </p>
        
        
        <h2>
          7 ways to survive in the desert
        </h2>
        
        
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.
        </p>
        
    
      </div>
      

    </article>
    

    <article>
      
    
      <img
        src="https://picsum.photos/id/37/800/300"
      />
      
    
      <div
        class="content"
      >
        
        
        <p
          class="category"
        >
          travel
        </p>
        
        
        <h2>
          5 amazing travel destinations
        </h2>
        
        
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.
        </p>
        
    
      </div>
      

    </article>
    

  </main>
</body>
`;
